apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/commons/QueryUtilsTest.java

```diff
@@ -128,6 +132,83 @@ public void testCreateDateRangePredicateBothBounds() {
     verify(builder).and(notNullPredicate, lowerBoundPredicate, upperBoundPredicate);
   }
 
+  /**
+   * Test {@link QueryUtils#hasClaimsData} asserting various ways an integer mask value can be
+   * created (mathematical XOR or additive integers) and result in the same integer value. The test
+   * further verifies that leveraging operator AND functionality to extract a subset of the bits in
+   * the state, one can arrive at a boolean true/false outcome.
+   *
+   * <p>The {@link QueryUtils#hasClaimsData} method takes in an arithmetic integer and returns a
+   * {@link BitSet} object that denotes availability of claims data, using predefined {@link
+   * QueryUtils} bitset identifiers.
+   */
+  @Test
+  public void testHasClaimsDataMaskValues() {
+    // 0001001 testVal
+    // 0000001 maskVal
+    // -------
+    // 0000001 expected
+    int testVal = QueryUtils.V_CARRIER_HAS_DATA | QueryUtils.V_SNF_HAS_DATA;
+    assertEquals(QueryUtils.V_CARRIER_HAS_DATA, (testVal & QueryUtils.V_CARRIER_HAS_DATA));
+
+    // same test as previous, but use additive additive values instead of XOR
+    testVal = QueryUtils.V_CARRIER_HAS_DATA + QueryUtils.V_SNF_HAS_DATA;
+    assertEquals(QueryUtils.V_CARRIER_HAS_DATA, (testVal & QueryUtils.V_CARRIER_HAS_DATA));
+
+    // 0001001 testVal
+    // 0010000 maskVal
+    // -------
+    // 0000000 expected
+    assertEquals(0, (testVal & QueryUtils.V_DME_HAS_DATA));
+
+    // test additive vs. XOR of values by comparing two integer values, each derived
+    // in a different manner.
+    testVal = QueryUtils.V_DME_HAS_DATA | QueryUtils.V_SNF_HAS_DATA | QueryUtils.V_HHA_HAS_DATA;
+    assertEquals(
+        testVal, QueryUtils.V_DME_HAS_DATA + QueryUtils.V_SNF_HAS_DATA + QueryUtils.V_HHA_HAS_DATA);
+
+    // test entire mask being set; each '&' mask should produce a value equal
+    // to mask we are checking for.
+    testVal =
+        QueryUtils.V_CARRIER_HAS_DATA
+            + QueryUtils.V_INPATIENT_HAS_DATA
+            + QueryUtils.V_OUTPATIENT_HAS_DATA
+            + QueryUtils.V_SNF_HAS_DATA
+            + QueryUtils.V_DME_HAS_DATA
+            + QueryUtils.V_HHA_HAS_DATA
+            + QueryUtils.V_HOSPICE_HAS_DATA
+            + QueryUtils.V_PART_D_HAS_DATA;
+
+    assertEquals(QueryUtils.V_PART_D_HAS_DATA, (testVal & QueryUtils.V_PART_D_HAS_DATA));
+    assertEquals(QueryUtils.V_HOSPICE_HAS_DATA, (testVal & QueryUtils.V_HOSPICE_HAS_DATA));
+    assertEquals(QueryUtils.V_HHA_HAS_DATA, (testVal & QueryUtils.V_HHA_HAS_DATA));
+    assertEquals(QueryUtils.V_DME_HAS_DATA, (testVal & QueryUtils.V_DME_HAS_DATA));
+    assertEquals(QueryUtils.V_SNF_HAS_DATA, (testVal & QueryUtils.V_SNF_HAS_DATA));
+    assertEquals(QueryUtils.V_OUTPATIENT_HAS_DATA, (testVal & QueryUtils.V_OUTPATIENT_HAS_DATA));
+    assertEquals(QueryUtils.V_INPATIENT_HAS_DATA, (testVal & QueryUtils.V_INPATIENT_HAS_DATA));
+    assertEquals(QueryUtils.V_CARRIER_HAS_DATA, (testVal & QueryUtils.V_CARRIER_HAS_DATA));
+
+    testVal =
```

Comment:
I feel as though everything above this line in the test may not be necessary - because the code above is testing the functionality of `&` and `+` for bit manipulation, and I feel like it's safe to assume that neither of those operators in Java are going to change in the future.

Reply:
True, but I wanted to emphasize that the mask returns the mask value (i.e., not 0, not 1 (unless CARRIER, etc.)

Reply:
removed some tests.